app/controllers/devise_registration_controller.rb

```diff
@@ -102,7 +120,23 @@ def phone_verify
   def phone_resend
     redirect_to url_for_state and return unless registration_state.state == "phone"
 
-    @phone = registration_state.phone
+    @phone = MultiFactorAuth.e164_number(params[:phone] || registration_state.phone)
+  end
+
+  def phone_resend_code
```

Comment:
Missing a `redirect_to url_for_state and return unless registration_state.state == "phone"`